src/rocksdb/src/lib.rs

```diff
@@ -103,6 +164,24 @@ impl InstanceOptions {
         // Set the env first so tuning applies to the shared `Env`.
         options.set_env(&self.env);
 
+        if let Some((fn_name, merge_fn)) = &self.merge_operator {
+            let bincode = self.bincode.clone();
+            let merge_fn = merge_fn.clone();
+            // We use an associative merge operator which is used for both full/partial merges
+            // since the value type is always the same for puts and merges.
+            // See `https://github.com/facebook/rocksdb/wiki/Merge-Operator#associativity-vs-non-associativity`
+            // for more info.
+            options.set_merge_operator_associative(fn_name, move |key, existing, operands| {
+                let existing = existing.map(|v| bincode.deserialize(v).unwrap());
+                let operands = ValueIterator {
+                    iter: operands.iter(),
+                    bincode: &bincode,
+                    v: std::marker::PhantomData::<V>,
+                };
+                merge_fn(key, existing, operands).map(|result| bincode.serialize(&result).unwrap())
```

Comment:
```suggestion
                let operands = ValueIterator {
                    iter: existing.chain(operands.iter()),
                    bincode: &bincode,
                    v: std::marker::PhantomData::<V>,
                };
                merge_fn(key, operands).map(|result| bincode.serialize(&result).unwrap())
```

could simplify the api, but not blocking